rclcpp/include/rclcpp/any_executable.hpp

```diff
@@ -26,8 +26,22 @@ namespace rclcpp
 namespace executor
 {
 
-struct AnyExecutable
+struct AnyExecutableState
 {
+  // Only one of the following pointers will be set.
+  rclcpp::subscription::SubscriptionBase * subscription;
+  rclcpp::subscription::SubscriptionBase * subscription_intra_process;
+  rclcpp::timer::TimerBase * timer;
+  rclcpp::service::ServiceBase * service;
+  rclcpp::client::ClientBase * client;
```

Comment:
How can we ensure that the pointers stored in here are still valid when the executable is actually being processed? The instance might have been deleted by then.